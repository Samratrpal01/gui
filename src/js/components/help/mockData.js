export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
