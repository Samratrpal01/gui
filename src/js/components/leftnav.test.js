import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import LeftNav from './leftnav';
import { defaultState, undefineds } from '../../../tests/mockData';
import { render } from '../../../tests/setupTests';

const mockStore = configureStore([thunk]);

describe('LeftNav Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <Provider store={store}>
        <LeftNav />
      </Provider>
    );
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
