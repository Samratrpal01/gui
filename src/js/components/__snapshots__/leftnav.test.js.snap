// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav Component renders correctly 1`] = `
.emotion-0 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  box-sizing: border-box;
  text-align: left;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-1.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-1.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-1.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-1.Mui-disabled {
  opacity: 0.38;
}

.emotion-2 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: block;
}

.emotion-18 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  min-width: 0;
  margin-top: 6px;
  margin-bottom: 6px;
}

.emotion-20 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.6);
  display: block;
}

<div>
  <ul
    class="MuiList-root MuiList-padding emotion-0"
    style="padding: 0px;"
  >
    <a
      aria-current="page"
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-1 active"
      href="/"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root emotion-2"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
        >
          Dashboard
        </span>
      </div>
    </a>
    <a
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-1"
      href="/devices"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root emotion-2"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
        >
          Devices
        </span>
      </div>
    </a>
    <a
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-1"
      href="/releases"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root emotion-2"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
        >
          Releases
        </span>
      </div>
    </a>
    <a
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-1"
      href="/deployments"
      style="padding: 22px 16px 22px 42px;"
    >
      <div
        class="MuiListItemText-root emotion-2"
        style="text-transform: uppercase;"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
        >
          Deployments
        </span>
      </div>
    </a>
  </ul>
  <ul
    class="MuiList-root MuiList-padding emotion-0"
    style="padding: 0px; position: absolute; bottom: 30px; left: 0px; right: 0px;"
  >
    <a
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding navLink leftNav emotion-1"
      href="/help"
      style="padding: 16px 16px 16px 42px;"
    >
      <div
        class="MuiListItemText-root emotion-2"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
        >
          Help
        </span>
      </div>
    </a>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding Mui-disabled emotion-1"
      disabled=""
      style="padding: 16px 16px 16px 42px; color: rgb(148, 148, 149);"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline emotion-18"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary emotion-3"
        >
          <div
            class="clickable"
            data-mui-internal-clone-element="true"
          />
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary emotion-20"
        >
          <a
            href="https://docs.mender.io/development/release-information/open-source-licenses"
            rel="noopener noreferrer"
            style="font-size: 13px; position: relative; top: 6px; color: rgb(25, 118, 210);"
            target="_blank"
          >
            License information
          </a>
        </p>
      </div>
    </li>
  </ul>
</div>
`;
